src/lib.rs

```diff
@@ -145,11 +148,44 @@ impl Exporter {
     /// Spawn exporter
     pub async fn new(options: ExporterOptions) -> Result<Self, Box<dyn std::error::Error>> {
         let connection_pool = SingleNodeConnectionPool::new(options.elasticsearch_url.clone());
-        let transport = TransportBuilder::new(connection_pool)
-            .timeout(options.elasticsearch_global_timeout)
-            .build()?;
 
-        let client = Elasticsearch::new(transport);
+        let mut transport =
+            TransportBuilder::new(connection_pool).timeout(options.elasticsearch_global_timeout);
+
+        let load_cert = || -> Result<Certificate, elasticsearch::Error> {
+            if let Some(ref cert_path) = options.elasticsearch_certificate_path {
+                let mut buf = Vec::new();
+                let _ = File::open(cert_path)?.read_to_end(&mut buf)?;
+                Certificate::from_pem(&buf)
+            } else {
+                panic!("Please provide --elasticsearch_certificate_path=CERTIFICATE_PATH flag");
```

Comment:
panicing as the user forgot to provide this flag when full or partial certificate mode is used